File: Jewels and Stones/Leetcode_771.cpp
class Solution {
public:
    int numJewelsInStones(string J, string S) {
        int ans = 0;
        unordered_set<char> jewels(J.begin(), J.end());
        for (char s: S)
            if (jewels.count(s)) ans++;
        return ans;
    }
};

File: Jewels and Stones/Leetcode_771.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        Set<String> jewels = new HashSet<String>();
        for (int i=0; i<J.length(); i++){
            jewels.add(J.substring(i, i+1));
        }
        int res=0;
        for(int i=0; i<S.length(); i++){
            if (jewels.contains(S.substring(i, i+1))) res++;
        }
        return res;
    }
}
